packages/phone-number-privacy/combiner/src/match-making/get-contact-matches.ts

```diff
@@ -66,12 +171,26 @@ export async function handleGetContactMatches(
   }
 }
 
+async function verifySignedUserPhoneNumberSignature(
+  account: string,
+  userPhoneNumber: string,
+  signedUserPhoneNumber: string,
+  logger: Logger
+) {
+  let dek
+  try {
+    dek = await getDataEncryptionKey(account, getContractKit(), logger)
+  } catch (error) {
+    throw new Error(ErrorMessage.CONTRACT_GET_FAILURE)
+  }
+  return verifySignature(userPhoneNumber, signedUserPhoneNumber, dek)
```

Comment:
If the `elliptic` docs are correct, I believe this check will fail because the signature generation does not hash the input before doing to low level signing, while the verify function does hash the input before doing the low level verification. If this does work though, I must misunderstand something.